src/libcore/sync/atomic.rs

```diff
@@ -148,7 +148,9 @@ unsafe impl Sync for AtomicBool {}
 /// This type has the same in-memory representation as a `*mut T`.
 #[cfg(target_has_atomic = "ptr")]
 #[stable(feature = "rust1", since = "1.0.0")]
-#[repr(transparent)]
+#[cfg_attr(target_pointer_width = "16", repr(align(2)))]
```

Comment:
Do we support any platforms where size isn't already equal to alignment for `*mut`? This seems redundant to me, and sacrifices the `#[repr(transparent)]`, which is useful.

Reply:
I'm not aware of any but I don't know what platforms Rust might want to support in the future.